docs/DeveloperGuide.md

```diff
@@ -1,38 +1,228 @@
 # Developer Guide
 
+## Table of Contents
++ [Acknowledgements](#acknowledgements)
++ [Design](#design)
+   * [Architecture](#architecture)
+   * [UI Component](#ui-component)
+   * [Storage Component](#storage-component)
++ [Implementation](#implementation)
++ [Product Scope](#product-scope)
+   * [Target User Profile](#target-user-profile)
+   * [Value Proposition](#value-proposition)
++ [User Stories](#user-stories)
++ [Non-Functional Requirements](#non-functional-requirements)
++ [Glossary](#glossary)
++ [Instructions for Manual Testing](#instructions-for-manual-testing)
+
+
+
 ## Acknowledgements
 
 {list here sources of all reused/adapted ideas, code, documentation, and third-party libraries -- include links to the original source as well}
 
-## Design & implementation
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
 
+## Design 
+### Architecture
+
```

Comment:
Consider adding a class diagram to show how classes interact on the scale of the entire program.